docs: add 0.19 migration guide (#379)
nervosnetwork/lumos: website/docs/migrations/migrate-to-v0.19.md

# Migration to Lumos@0.19

> Migrate from lumos@0.18 or earlier to Lumos@0.19

Reviewer: ```suggestion
Migrate from `lumos@0.18` or earlier to `lumos@0.19`
```

1. changed all snake_case to camelCase

> 2. removed the ckb-js-toolkit and replaced with @lumos/codec

Reviewer: ```suggestion
2. deprecated the `@ckb-lumos/toolkit` and replaced with `@ckb-lumos/codec`
3. deprecated `readBigUintX` in `@ckb-lumos/base`
```

# Migration to Lumos@0.19

Migrate from `lumos@0.18` or earlier to `lumos@0.19`

1. changed all snake_case to camelCase
2. deprecated the `@ckb-lumos/toolkit` and replaced with `@ckb-lumos/codec`
3. deprecated `readBigUintX` in `@ckb-lumos/base`

## Step-by-Step Guide

### Step 1

> Download the migration script and run it to automatically change your snake-case code to camel-case. Before doing this, you should commit your code to git and keep repo clean with no staging changes.

Reviewer: maybe we should make the warning first to tell user running the codemod will change their code

<pre>
### Step 1 - Run the Codemod

**WARNING**: Running the codemod will change your source code, please make sure your code are has been committed

Running the codemod will

1. change your lumos-related API from snake_case to camelCase
2. warning for deprecated function

```sh
curl -s https://gist.githubusercontent.com/zhangyouxin/e5ddf9b966f611173a01d6c98715c931/raw/30606c12756a866437a390add8dc5f5df47c9c36/codemod.js \
| xargs -0 -I{} node -e {} "your-source-dir/**/*.js"
```
</pre>

```sh
curl -s https://gist.githubusercontent.com/zhangyouxin/e5ddf9b966f611173a01d6c98715c931/raw/30606c12756a866437a390add8dc5f5df47c9c36/codemod.js \
| xargs -0 -I{} node -e {} "your-source-dir/**/*.js"
```

The output shoul look like: ![output](../../assets/codemod-ouput.png)

Then check out the changes by `git status` and `git diff` and make sure the code diff is within expectation.

The warnings logged in console indicates that you should take care of some deprecated functions your code is using, follow this guide to upgrade your code.

**Tips: If you have persisted data, remember to change snake-case to camel-case manually.**

> ### Step 2

Reviewer: ```suggestion
### Step 2 - (De)Serialization
```

Rewrite Serialization and DeSerialization codes.

```diff
+ import { blockchain } from "@ckb-lumos/base"
+ import { bytes } from "@ckb-lumos/codec"
// Deserialization
       const witness = '0x...'
-      const witnessArgs = new core.WitnessArgs(new toolkit.Reader(witness));
-      const lock = witnessArgs.getLock();
+      const witnessArgs = blockchain.WitnessArgs.unpack(bytes.bytify(witness))
+      const lock = witnessArgs.lock;

// Serialization:
     const newWitnessArgs = { lock: '0x...' };
-    witness = new toolkit.Reader(
-      core.SerializeWitnessArgs(toolkit.normalizers.NormalizeWitnessArgs(newWitnessArgs))
-    ).serializeJson();
+    witness = bytes.hexify(blockchain.WitnessArgs.pack(newWitnessArgs))
```

> ### Step 3

Reviewer: ```suggestion
### Step 3 - Utility Functions in `@ckb-lumos/base/utils`
```
